resources/lib/downloadutils.py

```diff
@@ -327,19 +329,25 @@ def authenticate(self):
         resp = self.downloadUrl(url, postBody=messageData, method="POST", suppress=True, authenticate=False)
 
         accessToken = None
+        userid = None
         try:
             result = json.loads(resp)
             accessToken = result.get("AccessToken")
+            userid = result["SessionInfo"].get("UserId")
         except:
             pass
 
         if accessToken is not None:
             log.debug("User Authenticated: {0}", accessToken)
             WINDOW.setProperty("AccessToken", accessToken)
+            WINDOW.setProperty("userid", userid)
+            WINDOW.setProperty("userimage", "")
```

Comment:
I don't think you need this

Reply:
userid has to be set there for hidden users, can't get the userid from the Public endpoint
and clearing the userimage to ensure there's no left over from changing users, since we check for the image in getuserid now and proceed to get image if unset and available

https://github.com/anxdpanic/plugin.video.embycon/blob/425cb28357707e9b64aa166d20a780a15706980f/resources/lib/downloadutils.py#L230-L233
if image is left set, we won't get the new image

Reply:
I mean just the line
WINDOW.setProperty("userimage", "")

Reply:
yea, it's there due to the lines mentioned, if you change users, the image would stay and not be updated otherwise

Reply:
is this okay?

Reply:
yeah, forgot about changing users, this would reset the user image and thus trigger a user image reset when getUserId was called next